src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2066,55 +2051,48 @@ private void ValidateOrganizationUpgradeParameters(Models.StaticStore.Plan plan,
             }
         }
 
-        private async Task ValidateOrganizationUserUpdatePermissionsAsync(Guid loggedInUserId, Guid organizationId,
-            OrganizationUserType newType, OrganizationUserType? oldType)
+        private void ValidateOrganizationUserUpdatePermissions(Guid organizationId, OrganizationUserType newType,
+            OrganizationUserType? oldType)
         {
-            var loggedInUserOrgs = await _organizationUserRepository.GetManyByUserAsync(loggedInUserId);
-            var loggedInAsOrgOwner = loggedInUserOrgs
-                .Any(u => u.OrganizationId == organizationId && u.Type == OrganizationUserType.Owner);
-            if (loggedInAsOrgOwner)
+            if (_currentContext.OrganizationOwner(organizationId))
             {
                 return;
             }
 
             var isOwner = oldType == OrganizationUserType.Owner;
             var nowOwner = newType == OrganizationUserType.Owner;
-            var ownerUserConfigurationAttempt = (isOwner && nowOwner) || !(isOwner.Equals(nowOwner));
+            var ownerUserConfigurationAttempt = (isOwner && nowOwner) || !isOwner.Equals(nowOwner);
```

Comment:
I wonder if we could simplify the logic here by doing `oldType == OrganizationUserType.Owner || newType == OrganizationUserType.Owner`.

Reply:
I like your approach better and seems to fit given the context and intent. Please go ahead and change this.